### PR TITLE
CI Setup

### PR DESCRIPTION
**Pull request by kchason**
Adds:

- `pre-commit` with CI checks -- note this has the unused imports ignored until impacts can be determined
- Poetry for installation and packaging
- CI for `pre-commit` and smoke test

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+[tool.poetry]
+name = "case-mapping"
```
Comment (ajnelson-nist) [inline pyproject.toml:2]: This name should be consistent with the primary source tree name, which at the moment is `case_builder`.  Should that directory be `case_mapping` instead?

IIRC, the hyphen here is appropriate.

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+[tool.poetry]
```
Comment (ajnelson-nist) [inline pyproject.toml:1]: This is new to me - does `poetry` need to be mentioned as the first line?

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    "Development Status :: 3 - Alpha",
+    "Enviroment :: Console",
+    "License :: OSI Approved :: Apache Software License",
+    "Operating System :: OS Independent",
+    "Programming Language :: Python :: 3",
+    "Topic :: Software Development :: Libraries"
+]
+
+[tool.poetry.dependencies]
+python = "^3.9"
```
Comment (ajnelson-nist) [inline pyproject.toml:22]: Is this how Python >= 3.9 is supposed to be spelled?

Are we supposed to enumerate Python versions in `pyproject.toml` as well as `ci.yml`?

File: pyproject.toml [hunk 4]
```diff
@@ -0,0 +1,27 @@
+[tool.poetry]
+name = "case-mapping"
+version = "0.1.0"
+description = "Classes to support representation of the CASE Cyber Ontology"
+authors = []
```
[the remark this answers is not in the document]
Comment (ajnelson-nist) [reply pyproject.toml:5]: Whoops, I meant to just annotate the `authors` line with this remark, sorry.

File: pyproject.toml [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+license = "Apache-2.0"
+readme = "README.md"
+repository = "https://github.com/casework/CASE-Mapping-Python"
+packages = [
+    { include = "case_builder" }
```
[the remark this answers is not in the document]
Comment (ajnelson-nist) [reply pyproject.toml:10]: Whoops, I meant to annotate just the `packages` line, sorry.

**Comment by ajnelson-nist** [comment]
I am fine with merging this failing CI if it appears CI fails only due to #2 .

Alternatively, #2 could be fixed by adding calls to `uuid.uuid4` in a separate PR.

File: pyproject.toml (hunk 2 above)
Comment (kchason) [reply pyproject.toml:1]: Yes, this is the template generated from `poetry init`

File: pyproject.toml (hunk 3 above)
Comment (kchason) [reply pyproject.toml:22]: This was defined during the `poetry init` process but it appears to build fine without it. Removing.

File: pyproject.toml (hunk 5 above)
[the remark this answers is not in the document]
Comment (ajnelson-nist) [reply pyproject.toml:10]: I think the change now is worthwhile.

We're already going to impact current consumers by switching from where the package comes.

Other public CASE repos using this code style do not run an import at all.

So, it seems now's the time to make a change.

File: pyproject.toml (hunk 4 above)
[the remark this answers is not in the document]
Comment (kchason) [reply pyproject.toml:5]: Adding `"Cyber Domain Ontology Maintainers <operations@cyberdomainontology.org>"`

File: pyproject.toml (hunk 1 above)
Comment (kchason) [reply pyproject.toml:2]: Removed with the directory rename below

File: pyproject.toml [hunk 6]
```diff
@@ -0,0 +1,27 @@
+[tool.poetry]
+name = "case-mapping"
+version = "0.1.0"
```
[the remark this answers is not in the document]
Comment (ajnelson-nist) [reply pyproject.toml:3]: We have some options available to us on how to handle this.  For reference, `case_mapping.__version__` is one way to get versions that other CASE projects currently support.

**Comment by kchason** [comment]
Merging to add `pre-commit` and other benefits, knowing that it will fail the validation until #2 is resolved

**Pull request merged**